Domains: Show "Free with your plans" for incoming transfer products

hambai opened the pull request:
Incoming transfers are also free with a plan, so this PR should display the "Free with your plan" notice next to the Domain Transfer and Domain Transfer Privacy products in the cart.

before:
<img width="273" alt="screen shot 2017-11-22 at 15 20 13" src="https://user-images.githubusercontent.com/1355045/33149340-14a1a8ba-cfd8-11e7-83fe-bd634ce86722.png">

after:
<img width="274" alt="screen shot 2017-11-22 at 22 54 31" src="https://user-images.githubusercontent.com/1355045/33149397-409f94ea-cfd8-11e7-9152-7f5aec513943.png">

Also I've fixed the Privacy Protection card on the checkout screen:
<img width="722" alt="screen shot 2017-11-22 at 22 55 00" src="https://user-images.githubusercontent.com/1355045/33149402-46e1b77a-cfd8-11e7-9d51-362c28134dde.png">

klimeryk commented on client/lib/products-values/index.js:
```diff
@@ -249,6 +253,13 @@ export function isDomainTransfer( product ) {
+	product = formatProduct( product );
+	assertValidProduct( product );
+
+	return product.product_slug === 'domain_transfer_privacy';
```
This should be the `TRANSFER_IN_PRIVACY` constant from `lib/domains/constants.js`.

klimeryk commented on client/lib/products-values/index.js:
```diff
@@ -210,7 +210,11 @@ export function isDomainProduct( product ) {
+		isDomainRegistration( product ) ||
+		isPrivacyProtection( product ) ||
+		isDomainTransfer( product ) ||
+		isDomainTransferPrivacy( product )
```
Seems like we don't need the check [here](https://github.com/Automattic/wp-calypso/blob/dc3426b6bf30be7eb62734b6d92d043bb3107bcd/client/me/purchases/purchase-item/index.jsx#L154) if transfers in are a domain product.

aidvu replied:
Not the same atm: https://github.com/Automattic/wp-calypso/blob/5f15f3365213253724ca0d8798c6cb8160f00bc9/client/lib/products-values/index.js#L208

We can review and see if it makes sense to set it as a domain product.

hambai commented:
OK, so I removed the domain transfer products from the `isDomainProduct` check - it makes sense to treat those as separate products. I fixed my code accordingly. I also took the liberty to refactor and remove the `isTransfer` function as it seems something that we should check in `product-values`.

aidvu commented:
Hmmm... Nevermind my comment. Somehow I didn't see the free-with-plan. Works ok now.
